React-native failed to build

## Describe the bug

All of the sudden from today react-native failed to build our mobile app

```
yarn run v1.22.19
$ react-native run-android
info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
Jetifier found 1423 file(s) to forward-jetify. Using 10 workers...
info JS server already running.
info Installing the app...

> Configure project :app
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed in version 5.0 of the Android Gradle plugin.
For more information, see http://d.android.com/r/tools/update-dependency-configurations.html.
[Bugsnag] "bugsnag-react-native.gradle" should not applied manually. Install the Bugsnag Android Gradle Plugin and remove this from your Gradle file.

> Configure project :realm

> Task :app:generatePackageList
> Task :app:preBuild
> Task :app:preDebugBuild
> Task :bugsnag_react-native:preBuild UP-TO-DATE
> Task :bugsnag_react-native:preDebugBuild UP-TO-DATE
> Task :bugsnag_react-native:compileDebugAidl NO-SOURCE
> Task :react-native-async-storage_async-storage:preBuild UP-TO-DATE
> Task :react-native-async-storage_async-storage:preDebugBuild UP-TO-DATE
> Task :react-native-async-storage_async-storage:compileDebugAidl NO-SOURCE
> Task :react-native-ble-plx:preBuild UP-TO-DATE
> Task :react-native-ble-plx:preDebugBuild UP-TO-DATE
> Task :react-native-ble-plx:compileDebugAidl NO-SOURCE
> Task :react-native-bluetooth-status:preBuild UP-TO-DATE
> Task :react-native-bluetooth-status:preDebugBuild UP-TO-DATE
> Task :react-native-bluetooth-status:compileDebugAidl NO-SOURCE
> Task :react-native-camera:preBuild UP-TO-DATE
> Task :react-native-camera:preGeneralDebugBuild UP-TO-DATE
> Task :react-native-camera:compileGeneralDebugAidl NO-SOURCE
> Task :react-native-community_checkbox:preBuild UP-TO-DATE
> Task :react-native-community_checkbox:preDebugBuild UP-TO-DATE
> Task :react-native-community_checkbox:compileDebugAidl NO-SOURCE
> Task :react-native-community_datetimepicker:preBuild UP-TO-DATE
> Task :react-native-community_datetimepicker:preDebugBuild UP-TO-DATE
> Task :react-native-community_datetimepicker:compileDebugAidl NO-SOURCE
> Task :react-native-community_masked-view:preBuild UP-TO-DATE
> Task :react-native-community_masked-view:preDebugBuild UP-TO-DATE
> Task :react-native-community_masked-view:compileDebugAidl NO-SOURCE
> Task :react-native-community_picker:preBuild UP-TO-DATE
> Task :react-native-community_picker:preDebugBuild UP-TO-DATE
> Task :react-native-community_picker:compileDebugAidl NO-SOURCE
> Task :react-native-community_slider:preBuild UP-TO-DATE
> Task :react-native-community_slider:preDebugBuild UP-TO-DATE
> Task :react-native-community_slider:compileDebugAidl NO-SOURCE
> Task :react-native-device-info:preBuild UP-TO-DATE
> Task :react-native-device-info:preDebugBuild UP-TO-DATE
> Task :react-native-device-info:compileDebugAidl NO-SOURCE
> Task :react-native-document-picker:preBuild UP-TO-DATE
> Task :react-native-document-picker:preDebugBuild UP-TO-DATE
> Task :react-native-document-picker:compileDebugAidl NO-SOURCE
> Task :react-native-fs:preBuild UP-TO-DATE
> Task :react-native-fs:preDebugBuild UP-TO-DATE
> Task :react-native-fs:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:preBuild UP-TO-DATE
> Task :react-native-gesture-handler:preDebugBuild UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugAidl NO-SOURCE
> Task :react-native-haptic-feedback:preBuild UP-TO-DATE
> Task :react-native-haptic-feedback:preDebugBuild UP-TO-DATE
> Task :react-native-haptic-feedback:compileDebugAidl NO-SOURCE
> Task :react-native-localization:preBuild UP-TO-DATE
> Task :react-native-localization:preDebugBuild UP-TO-DATE
> Task :react-native-localization:compileDebugAidl NO-SOURCE
> Task :react-native-mail:preBuild UP-TO-DATE
> Task :react-native-mail:preDebugBuild UP-TO-DATE
> Task :react-native-mail:compileDebugAidl NO-SOURCE
> Task :react-native-permissions:preBuild UP-TO-DATE
> Task :react-native-permissions:preDebugBuild UP-TO-DATE
> Task :react-native-permissions:compileDebugAidl NO-SOURCE
> Task :react-native-restart:preBuild UP-TO-DATE
> Task :react-native-restart:preDebugBuild UP-TO-DATE
> Task :react-native-restart:compileDebugAidl NO-SOURCE
> Task :react-native-safe-area-context:preBuild UP-TO-DATE
> Task :react-native-safe-area-context:preDebugBuild UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugAidl NO-SOURCE
> Task :react-native-screens:preBuild UP-TO-DATE
> Task :react-native-screens:preDebugBuild UP-TO-DATE
> Task :react-native-screens:compileDebugAidl NO-SOURCE
> Task :react-native-svg:preBuild UP-TO-DATE
> Task :react-native-svg:preDebugBuild UP-TO-DATE
> Task :react-native-svg:compileDebugAidl NO-SOURCE
> Task :react-native-system-setting:preBuild UP-TO-DATE
> Task :react-native-system-setting:preDebugBuild UP-TO-DATE
> Task :react-native-system-setting:compileDebugAidl NO-SOURCE
> Task :react-native-vector-icons:preBuild UP-TO-DATE
> Task :react-native-vector-icons:preDebugBuild UP-TO-DATE
> Task :react-native-vector-icons:compileDebugAidl NO-SOURCE
> Task :realm:send
> Task :realm:preBuild
> Task :realm:preDebugBuild
> Task :realm:compileDebugAidl NO-SOURCE
> Task :app:compileDebugAidl NO-SOURCE
> Task :bugsnag_react-native:packageDebugRenderscript NO-SOURCE
> Task :react-native-async-storage_async-storage:packageDebugRenderscript NO-SOURCE
> Task :react-native-ble-plx:packageDebugRenderscript NO-SOURCE
> Task :react-native-bluetooth-status:packageDebugRenderscript NO-SOURCE
> Task :react-native-camera:packageGeneralDebugRenderscript NO-SOURCE
> Task :react-native-community_checkbox:packageDebugRenderscript NO-SOURCE
> Task :react-native-community_datetimepicker:packageDebugRenderscript NO-SOURCE
> Task :react-native-community_masked-view:packageDebugRenderscript NO-SOURCE
> Task :react-native-community_picker:packageDebugRenderscript NO-SOURCE
> Task :react-native-community_slider:packageDebugRenderscript NO-SOURCE
> Task :react-native-device-info:packageDebugRenderscript NO-SOURCE
> Task :react-native-document-picker:packageDebugRenderscript NO-SOURCE
> Task :react-native-fs:packageDebugRenderscript NO-SOURCE
> Task :react-native-gesture-handler:packageDebugRenderscript NO-SOURCE
> Task :react-native-haptic-feedback:packageDebugRenderscript NO-SOURCE
> Task :react-native-localization:packageDebugRenderscript NO-SOURCE
> Task :react-native-mail:packageDebugRenderscript NO-SOURCE
> Task :react-native-permissions:packageDebugRenderscript NO-SOURCE
> Task :react-native-restart:packageDebugRenderscript NO-SOURCE
> Task :react-native-safe-area-context:packageDebugRenderscript NO-SOURCE
> Task :react-native-screens:packageDebugRenderscript NO-SOURCE
> Task :react-native-svg:packageDebugRenderscript NO-SOURCE
> Task :react-native-system-setting:packageDebugRenderscript NO-SOURCE
> Task :react-native-vector-icons:packageDebugRenderscript NO-SOURCE
> Task :realm:packageDebugRenderscript NO-SOURCE
> Task :app:compileDebugRenderscript NO-SOURCE
> Task :app:generateDebugBuildConfig UP-TO-DATE
> Task :app:javaPreCompileDebug UP-TO-DATE
> Task :bugsnag_react-native:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-async-storage_async-storage:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-ble-plx:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-bluetooth-status:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-camera:writeGeneralDebugAarMetadata UP-TO-DATE
> Task :react-native-community_checkbox:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-community_datetimepicker:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-community_masked-view:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-community_picker:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-community_slider:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-device-info:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-document-picker:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-fs:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-gesture-handler:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-haptic-feedback:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-localization:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-mail:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-permissions:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-restart:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-safe-area-context:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-screens:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-svg:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-system-setting:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-vector-icons:writeDebugAarMetadata UP-TO-DATE
> Task :realm:writeDebugAarMetadata UP-TO-DATE
> Task :app:bundleDebugJsAndAssets SKIPPED
> Task :app:checkDebugAarMetadata FAILED
> Task :app:mergeDebugShaders UP-TO-DATE

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings
31 actionable tasks: 3 executed, 28 up-to-date
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/03 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/03 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/03 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugAarMetadata'.
> Multiple task action failures occurred:
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.appcompat:appcompat:1.4.1.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/3aeebd481c4b979f4aaf1422ab700cb2/appcompat-1.4.1/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.appcompat:appcompat-resources:1.4.1.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/5037444d0e4e7ecd374e14c9cde1a9e9/jetified-appcompat-resources-1.4.1/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.emoji2:emoji2-views-helper:1.0.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/42e939cb6ff04968d17f51f220845a50/jetified-emoji2-views-helper-1.0.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.emoji2:emoji2:1.0.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/6d54bbfebc644b955342f5222212631f/jetified-emoji2-1.0.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.lifecycle:lifecycle-process:2.4.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/b4f70e2aa5cce10e50e20d9b8e994481/jetified-lifecycle-process-2.4.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.tracing:tracing:1.1.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/87def6e2f368a548bd627f1e462ca7d3/jetified-tracing-1.1.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.core:core:1.7.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/e596bd84a87a4cbdcf9d545f7eacbd1e/core-1.7.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.lifecycle:lifecycle-runtime:2.4.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/7e88b296a48384d725962d71d07ec64e/lifecycle-runtime-2.4.0/META-INF/com/android/build/gradle/aar-metadata.properties.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/03 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/03 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/03 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugAarMetadata'.
> Multiple task action failures occurred:
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.appcompat:appcompat:1.4.1.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/3aeebd481c4b979f4aaf1422ab700cb2/appcompat-1.4.1/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.appcompat:appcompat-resources:1.4.1.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/5037444d0e4e7ecd374e14c9cde1a9e9/jetified-appcompat-resources-1.4.1/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.emoji2:emoji2-views-helper:1.0.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/42e939cb6ff04968d17f51f220845a50/jetified-emoji2-views-helper-1.0.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.emoji2:emoji2:1.0.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/6d54bbfebc644b955342f5222212631f/jetified-emoji2-1.0.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.lifecycle:lifecycle-process:2.4.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/b4f70e2aa5cce10e50e20d9b8e994481/jetified-lifecycle-process-2.4.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.tracing:tracing:1.1.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/87def6e2f368a548bd627f1e462ca7d3/jetified-tracing-1.1.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.core:core:1.7.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/e596bd84a87a4cbdcf9d545f7eacbd1e/core-1.7.0/META-INF/com/android/build/gradle/aar-metadata.properties.
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-29).
        Dependency: androidx.lifecycle:lifecycle-runtime:2.4.0.
        AAR metadata file: /Users/sussol/.gradle/caches/transforms-2/files-2.1/7e88b296a48384d725962d71d07ec64e/lifecycle-runtime-2.4.0/META-INF/com/android/build/gradle/aar-metadata.properties.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s

    at makeError (/Users/sussol/Documents/Projects/mobile/node_modules/execa/index.js:174:9)
    at /Users/sussol/Documents/Projects/mobile/node_modules/execa/index.js:278:16
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async runOnAllDevices (/Users/sussol/Documents/Projects/mobile/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:94:5)
    at async Command.handleAction (/Users/sussol/Documents/Projects/mobile/node_modules/@react-native-community/cli/build/index.js:186:9)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### To reproduce

Steps to reproduce the behaviour:

1. Try to build the application to run in development mode
2. Open a terminal and do `yarn start` 
3. Open another terminal and do `yarn android`

### Expected behaviour

- React native should build properly without error

### Proposed Solution

Android studio is trying to enforce version 31 of development kit to our app. However we cannot use version 31 as it is not supported by realm. 
Update build settings.

### Version and device info

- App version: v8.6.1
- Tablet model: 
- OS version:

### Additional context

Add any other context about the problem here.

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -21,18 +21,16 @@ buildscript {
     }
 }
 
+def REACT_NATIVE_VERSION = new File(['node', '--print',"JSON.parse(require('fs').readFileSync(require.resolve('react-native/package.json'), 'utf-8')).version"].execute(null, rootDir).text.trim())
+
 allprojects {
-    repositories {
-        exclusiveContent {
-           filter {
-               includeGroup "com.facebook.react"
-           }
-           forRepository {
-               maven {
-                   url "$rootDir/../node_modules/react-native/android"
-               }
-           }
+    configurations.all {
+        resolutionStrategy {
+            // Remove this override in 0.65+, as a proper fix is included in react-native itself.
+            force "com.facebook.react:react-native:" + REACT_NATIVE_VERSION
         }
+    }
+    repositories {
         mavenLocal()
         google()
         jcenter()

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -23,6 +23,16 @@ buildscript {
 
 allprojects {
     repositories {
+        exclusiveContent {
+           filter {
+               includeGroup "com.facebook.react"
+           }
+           forRepository {
+               maven {
+                   url "$rootDir/../node_modules/react-native/android"
+               }
+           }
+        }
         mavenLocal()
         google()
         jcenter()

